service/docs/source/security.rst

```diff
@@ -0,0 +1,525 @@
+documented, and the description includes the security implications of
+granting access. The ``geopmaccess`` command line tool is used by a
+system administrator to query descriptions of signals and controls
+available on a system as well as managing user access to each feature
+by name.
+
+
+Restore control configuration after a writing session ends
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Users are not allowed to make persistent changes to hardware settings
+using the GEOPM Service. The lifetime of configuration changes made
+through the GEOPM Service is limited to the lifetime of the writing
+PID's process session.  To learn more about the Linux process session
+interface see the
+`setsid(2) <https://man7.org/linux/man-pages/man2/setsid.2.html>`__
+man page.  Typically the process session is coupled to a controlling
+terminal.
+
+The GEOPM Service implements this feature by saving all control
+configuration settings to disk prior to enabling a user session to
+write. When the user's write session terminates, all configuration
+settings are restored to their previous value. The storage location
+for the saved state is in the directory
+``/var/run/geopm-service/SAVE_FILES``.
+
+Note that all configuration settings are restored regardless of
+whether or not they were adjusted during the session through the
+service.
+
+
+One writing GEOPM session allowed at any time
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+GEOPM Service enforces that at most one writing user session is
+supported at any time.  This restriction enables the save/restore
+methodology outlined in the previous section to be effective. The
+write lock that enables this mutual exclusion is associated with the
+Unix "process session leader" PID of the requesting process. If this
+session leader PID does not refer to an active process, then the lock
+is associated with the requesting process PID. Otherwise, any PID
+within the requesting Unix process session is enabled to write through
+the GEOPM Service for the duration of the process session leader PID
+lifetime.
+
+
+Provide a low latency interface
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The GEOPM Service provides a batch interface that allows users to
+configure a server process.  The batch server process is forked by
+``geopmd`` which is the primary GEOPM Service daemon process started
+by
+`systemd(1) <https://man7.org/linux/man-pages/man1/systemd.1.html>`__.
+The user configures the batch server with an array of requests through
+DBus. The access permissions are checked when the call to start the
+server is made, and the server is only forked if the client has
+permission for all requests in the array. The batch server creates one
+or two inter-process shared memory regions to interact with the
+client: one region provides a user interface to read signals, and the
+other if requested provides a user interface to write
+controls. Requests to read, write or quit are made through a FIFO
```

Comment:
```suggestion
controls. Requests to read, write, or quit are made through a FIFO
```

I can't remember if we do serial commas or not.  I think we waffle back and forth on that and one or two spaces after the period.